.harness/reconcile_dependencies_for_all_envs_based_on_parent_folder.yaml

```diff
@@ -0,0 +1,154 @@
+                      echo "=== === === === === === ==="
+
+
+                      # Iterate over each directory in the list
+                      for DIR in $CHANGED_DIRS; do
+                        # Run the script for each directory
+                        cd ${DIR} || exit 1
+                        echo "Running: bash ${ROOT_DIR}/tools/reconcile_dependencies.sh $DIR"
+                        bash ${ROOT_DIR}/tools/reconcile_dependencies.sh "$DIR"
+
+                        # This could be a separate step, but we already iterate over folders here.
+                        echo "Checking whether Environment Version ID should be updated"
+                        git show master:./env_info.json | grep -n 'environmentVersionId' > /tmp/old_env_version_id.txt
+                        git show HEAD:./env_info.json | grep -n 'environmentVersionId' > /tmp/new_env_version_id.txt
+                        # If env version was not updated, update
+                        if diff /tmp/old_env_version_id.txt /tmp/new_env_version_id.txt > /dev/null; then
+                          echo "Updating Environment Version ID"
+                          python3 ${ROOT_DIR}/tools/env_version_update.py --file ./env_info.json
+                        fi
+
+
+                        cd -
+                      done
+
+                      # Commit and push changes if any
+                      if [[ -n $(git status --porcelain) ]]; then
+                        git status --porcelain
+                        git commit -a -m "Reconcile dependencies for $ENV_DIR"
+                        git config pull.rebase true
+                        git pull origin "$(git branch --show-current)" --rebase
+                        git push --set-upstream origin "$(git branch --show-current)"
+
+                        echo "Reconciled dependencied commited to branch"
+                        exit 0
+                      else
+                        echo "No changes detected in Git."
+                      fi
+                    envVariables:
+                      GITHUB_ACCESS_TOKEN: <+secrets.getValue("account.githubpatsvcharnessgit2")>
+                    resources:
+                      limits:
+                        memory: 5Gi
+                  description: Run the tools/reconcile_dependencies.sh script to auto-generate requirements.txt file base on its requirements.in file.
+          platform:
+            os: Linux
+            arch: Amd64
+          runtime:
+            type: Cloud
+            spec: {}
+  variables:
+    - name: env_folder
+      type: String
+      description: The folder containing the environment to reconcile, e.g. "public_dropin_environments"
+      required: true
+      value: <+input>.allowedValues(public_dropin_environments)
+  identifier: reconcile_dependencies_for_all_envs_based_on_parent_folder
+  description: |-
+    On PR, for every drop in env defined in input set, check reconcile dependencies.
```

Comment:
PR check trigger and the inputset should be created? perhaps in followup PR?

Reply:
I didn't clean this part when cloned pipeline, I'll clean